### ISSUE TITLE
BME280 delivering wrong values

### ISSUE THREAD

**Issue by bgoll** [OttoWinter/esphomeyaml#109]
I try to use a BME280 via a Wemos D1 Mini. Here is my config:
```
i2c:
  sda: D2
  scl: D1
sensor:
  - platform: bme280
    temperature:
      name: "temperatur_bad"
      #oversampling: 16x
    pressure:
      name: "luftdruck_bad"
    humidity:
      name: "luftfeuchtigkeit_bad"
    address: 0x76
    update_interval: 10s
```

Flashing and setting up the sensor works without any problems, however the values received from the sensor are wrong:

```
[20:50:43][C][sensor.bme280:095]: Setting up BME280...
[20:50:43][C][sensor.bme280:096]:     Address: 0x76
[20:50:43][C][sensor.bme280:126]:     Temperature Oversampling: 16x
[20:50:43][C][sensor.bme280:127]:     Pressure Oversampling: 16x
[20:50:43][C][sensor.bme280:128]:     Humidity Oversampling: 16x
[20:50:43][C][sensor.bme280:129]:     IIR Filter: OFF
[20:50:43][C][component:239]:     Update interval: 10000ms
[20:50:43][C][sensor.mqtt:025]: Setting up MQTT Sensor 'luftfeuchtigkeit_bad'...
[20:50:43][C][sensor.mqtt:027]:     Expire After: 450s
[20:50:43][C][sensor.mqtt:029]:     Unit of Measurement: '%'
[20:50:43][C][sensor.mqtt:030]:     Accuracy Decimals: 1
[20:50:43][C][sensor.mqtt:031]:     Icon: 'mdi:water-percent'
[20:50:43][C][sensor.mqtt:025]: Setting up MQTT Sensor 'luftdruck_bad'...
[20:50:43][C][sensor.mqtt:027]:     Expire After: 450s
[20:50:43][C][sensor.mqtt:029]:     Unit of Measurement: 'hPa'
[20:50:43][C][sensor.mqtt:030]:     Accuracy Decimals: 1
[20:50:43][C][sensor.mqtt:031]:     Icon: 'mdi:gauge'
[20:50:43][C][sensor.mqtt:025]: Setting up MQTT Sensor 'temperatur_bad'...
[20:50:43][C][sensor.mqtt:027]:     Expire After: 450s
[20:50:43][C][sensor.mqtt:029]:     Unit of Measurement: '°C'
[20:50:43][C][sensor.mqtt:030]:     Accuracy Decimals: 1
[20:50:43][C][sensor.mqtt:031]:     Icon: ''
[20:50:43][I][application:090]: Running through first loop()
[20:50:43][I][application:106]: First loop finished successfully!
[20:50:43][D][sensor.bme280:193]: Got temperature=0.3°C pressure=313389.3hPa humidity=12.5%
[20:50:43][D][sensor.mqtt:091]: 'temperatur_bad': Pushing out value 0.280000 with 1 decimals of accuracy
[20:50:43][D][sensor.mqtt:091]: 'luftdruck_bad': Pushing out value 313389.312500 with 1 decimals of accuracy
[20:50:43][D][sensor.mqtt:091]: 'luftfeuchtigkeit_bad': Pushing out value 12.490234 with 1 decimals of accuracy
[20:50:49][D][sensor.bme280:193]: Got temperature=0.3°C pressure=313438.8hPa humidity=12.1%
[20:50:59][D][sensor.bme280:193]: Got temperature=0.3°C pressure=313557.8hPa humidity=12.1%
```
As you can see, e.g. the temperature is 0.3°C, which is definitely not true :-) The output continues in the same way. The humidity and pressure vary slightly, however the temperature is always exactly 0.3°C.

I verified that the sensor is working correctly and wired properly by running a short sketch on the board using this BME280 lib: https://github.com/finitespace/BME280
Here everything works without any problems:

```
Temp: 27.12°C		Humidity: 45.34% RH		Pressure: 98062.88Pa
```

**Comment by carlosmgr** [comment]
You just need to had this:

esphomeyaml:
  library_uri: 'https://github.com/exxamalte/esphomelib.git#i2c-fix'

**Comment by bgoll** [comment]
@carlosmgr works perfectly, thanks a lot! Any clue when this fix will be merged into esphomelib?

**Comment by sir106** [comment]
if this fix works - how about a pull request? didn‘t try it yet - any feedback?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by balk77** [comment]
It already has been fixed. I believe it will be part of the next issue. [#60](https://github.com/OttoWinter/esphomeyaml/issues/60)

**Comment by exxamalte** [comment]
Yes, this has been fixed and merged in https://github.com/OttoWinter/esphomelib/pull/98 back in June, but has not yet been officially released.

**Comment by OttoWinter** [comment]
Should be fixed in 1.8.0
